src/sentry/plugins/providers/integration_repository.py

```diff
@@ -102,13 +106,19 @@ def handle_api_error(self, error):
     def get_config(self, organization):
         raise NotImplementedError
 
-    def validate_config(self, organization, config):
+    def get_repository_data(self, organization, config):
+        """
+        Gets the necessary repository data through the integration's API
+        """
         return config
 
-    def create_repository(self, organization, data):
+    def build_repository_config(self, organization, data):
+        """
+        Builds final dict containing all necessary data to create the repository
```

Comment:
I think it could be pretty useful to add a sample of what this would look like since they all have same format afaik